### PR TITLE
Fix user credential update through PUT endpoint

### LINKED ISSUE #913: Fix user credential update through PUT endpoint
### Description

Updating user credential through "PUT /users/{id}" does not work. The credential is treated as a normal attribute and stored in the attributes column and returned in the response.

### Steps to Reproduce

1. Create a user (create an Ou and Schema also if not exists) with a password.
2. Update the user "PUT /users/{id}" providing a new password. Notice that the password attribute is there in the response body.
3. Try to authenticate the user with the new password. Notice that authentication fails. Authentication with old password will be successful.

### Version

v0.14.0

### Environment Details (with versions)

_No response_

### PR DESCRIPTION
**Pull request by nishagii**
### Purpose

Fixes #913 - User credentials (password) are now properly extracted, hashed, and stored separately when updating users via `PUT /users/{id}` endpoint.

**Problem:**
When updating a user with a password in the request body:
- ❌ Password was stored as plaintext in the `ATTRIBUTES` column
- ❌ Password was exposed in API responses (security vulnerability)
- ❌ Authentication with the updated password failed
- ❌ Old password continued to work after update

**Root Cause:**
The `UpdateUser` method was not extracting credentials from attributes before persisting to the database, causing credentials to remain in the attributes JSON instead of being hashed and stored in the `CREDENTIALS` column.

**Security Improvements:**
- ✅ Passwords hashed using PBKDF2 with salt before storage
- ✅ Credentials stored in `CREDENTIALS` column, not `ATTRIBUTES`
- ✅ Passwords never exposed in API responses
- ✅ Proper credential isolation from user attributes
- ✅ Existing credentials preserved during updates (merge, not replace)

### Approach

Modified the `UpdateUser` method in `backend/internal/user/service.go` to properly handle credentials by following the existing patterns used in `CreateUser`:

**Implementation Steps:**

1. **Extract credentials from attributes** using the existing `extractCredentials()` method
   - Removes credential fields (password, passkeys, etc.) from the attributes map
   - Hashes credentials using the configured hash service
   - Returns a `Credentials` map with hashed values

2. **Update user attributes** (without credentials) via `userStore.UpdateUser()`
   - Ensures attributes are persisted without sensitive credential data

3. **Merge and update credentials** via new `mergeAndUpdateCredentials()` helper method:
   - Retrieves existing credentials from database using `userStore.GetCredentials()`
   - Merges new credentials with existing ones using `mergeCredentials()` helper
   - Preserves existing credential types (e.g., passkeys remain when updating password)
   - Updates merged credentials in the `CREDENTIALS` column via `userStore.UpdateUserCredentials()`

**Key Design Decisions:**

- **Reused existing methods**: Leveraged `extractCredentials()` and `mergeCredentials()` to maintain DRY principles
- **Added helper method**: Created `mergeAndUpdateCredentials()` to encapsulate credential merging logic for better code organization and testability

**Changes Made:**

- `backend/internal/user/service.go`:
  - Updated `UpdateUser()` method (lines 381-424) to extract and handle credentials
  - Added `mergeAndUpdateCredentials()` helper method (lines 601-646) for credential merging logic

- `backend/internal/user/service_test.go`:
  - Added 9 comprehensive unit tests covering success paths, error handling, and edge cases
  - Defined test constants (`testUserID`, `testCustomerOUID`, `testCustomerType`) for reusability

### Related Issues

- Closes #913

### Related PRs

- N/A

### Checklist

- [x] Followed the contribution guidelines
- [x] Manual test round performed and verified
  - [x] Create user with password - password hashed correctly
  - [x] Update user with new password - password not in response, stored hashed
  - [x] Authenticate with new password - authentication succeeds
  - [x] Database verification - password in CREDENTIALS, not ATTRIBUTES
  - [x] Update user without password - credentials unchanged
- [x] Documentation provided
  - No documentation changes required - internal bug fix maintaining existing API contract
- [x] Tests provided
  - [x] Unit Tests - 9 tests cases achieving 100% coverage for new code
    - Success paths: credential extraction, hashing, merging, and updates without credentials
    - Error handling: user not found and database errors during retrieval and updates
    - Edge cases: multiple credential types, empty credentials after extraction
  - [ ] Integration Tests - N/A (unit tests provide sufficient coverage for this fix)
- [ ] Breaking changes
  - [ ] Breaking changes section filled - N/A
  - [ ] `breaking change` label added - N/A

**Test Results:**
-  All 82 tests pass
-  Linter passes with 0 issues

### Security checks

- [x] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets
  - Test data uses generic passwords only
  - No credentials hardcoded in implementation
  - All sensitive test data is local and disposable

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved credential handling during user updates with enhanced error handling and automatic transaction rollback on failure.
  * Ensured data consistency by merging credentials within a single transactional operation.

* **Tests**
  * Added comprehensive test coverage for credential update scenarios, error handling paths, and validation failures.

### CODE REVIEW

File: backend/internal/user/service.go [hunk 1]
```diff
@@ -401,25 +401,43 @@ func (us *userService) UpdateUser(ctx context.Context, userID string, user *User
 	// Ensure the user object has the correct ID
 	user.ID = userID
 
-	var updatedUser *User
+	if svcErr := us.validateOrganizationUnitForUserType(user.Type, user.OrganizationUnit, logger); svcErr != nil {
```
Comment (ThumulaPerera) [inline backend/internal/user/service.go]: Shall we move this into the transaction block?

Fundamentally, it feels okay to move both validations outside the transaction block as they are only doing READs. But as discussed with @JeethJJ, currently we are doing all DB operations (including READs) inside transaction blocks. He will initiate a discussion to revisit that decision. But until we come to a conclusion, lets stick to the current practice

Comment (JeethJJ) [reply backend/internal/user/service.go]: We decided to use transections for all database WRITE operations regardless its single or multiple operations at once. If there is a READ operation along with the WRITE operation (such as this case), we still include the read operation within the transection context too. Standalone READ operations can be independent from transections.
